### ISSUE TITLE
Parse error on @ patterns

### ISSUE THREAD

**Issue by nponeccop** [ndmitchell/hlint#277]
The bug is similar to other parse errors, and a workaround is to use -XHaskell2010.

A minimal example:

``` Haskell
foo x @ Nothing = 42
```

```
$ hlint aa.hs
aa.hs:2:1: Error: Parse error in pattern: @Nothing
Found:
    foo x @ Nothing = 42

  >
```

HLint 1.9.37

**Comment by mpickering** [comment]
This is because `TypeApplications` is turned on.

**Comment by nponeccop** [comment]
So what is recommended?

**Comment by ndmitchell** [comment]
I've disabled `TypeApplications` by default. I note that GHC parses the above fragment, so I consider this a haskell-src-exts bug.

**Comment by mpickering** [comment]
This program doesn't compile for me. Which program are you trying?

```
{-# LANGUAGE TypeApplications #-}

foo x @ Nothing = 42
```

**Comment by nponeccop** [comment]
This program is not supposed to compile. The problem was that implicit `-XTypedApplications` turned a pretty legal program (in my post, without any pragmas) into an illegal program in your post.

**Comment by ndmitchell** [comment]
@mpickering my bad - it doesn't compile with the extension enabled in Haskell either. Not an HSE bug.

@nponeccop, I would probably avoid writing that pattern with the spaces as a future GHC may break it.

Worked around by disabling TypeApplications (will be released in the next few days).

**Comment by ndmitchell** [comment]
Released version should no longer have problems with that file.
